packages/mui-utils/src/fastDeepAssign/fastDeepAssign.ts

```diff
@@ -0,0 +1,95 @@
+// Based on https://github.com/TehShrike/deepmerge
+// Based on https://github.com/fastify/deepmerge
+// MIT License
+// Copyright (c) 2012 - 2022 James Halliday, Josh Duff, and other contributors of deepmerge
+
+/* eslint-disable guard-for-in */
+/* eslint-disable no-else-return */
+
+/**
+ * Assigns props from one object to another. Focused on performance, only normal objects with no
+ * prototype are supported.
+ */
+export default function fastDeepAssign<T extends Record<string, any>>(target: T, source: any) {
+  const sourceIsArray = Array.isArray(source);
+  const targetIsArray = Array.isArray(target);
+
+  if (isPrimitive(source)) {
+    return source;
+  } else if (isPrimitiveOrBuiltIn(target)) {
```

Comment:
Why not just check for built in here? We have the primitive check already on the previous condition.

Reply:
I didn't write it, I pulled it from the source listed at the top and made a few modifications to simplify it. I could simplify it further.

Reply:
Ah, I thought is a new code, we can update it later in that case, up to you.